packages/mui-material/src/Button/Button.js

```diff
@@ -320,19 +349,38 @@ const ButtonStartIcon = styled('span', {
         marginLeft: -2,
       },
     },
+    {
+      props: { loadingPosition: 'start', loading: true },
+      style: {
+        transition: theme.transitions.create(['opacity'], {
+          duration: theme.transitions.duration.short,
+        }),
+        opacity: 0,
+      },
+    },
+    {
+      props: { loadingPosition: 'start', loading: true, fullWidth: true },
+      style: {
+        marginRight: -8,
```

Comment:
Are the margin values supposed to be the same as the icons? If so, can we add a regression test to check that, in the future, we don't change the icon's position without also updating the loading indicator position?

Reply:
> Are the margin values supposed to be the same as the icons?

Not really, it is for `fullWidth` case. The negative margin must be accounted for the loader size but it'd be too complex to add `calc()` with CSS variables at the moment, so I leave it like this to preserve the same behavior as the LoadingButton from the lab.